lbz/exceptions.py

```diff
@@ -37,83 +41,288 @@ def get_response(self, request_id: str) -> Response:
+    status_code = HTTPStatus.UNSUPPORTED_MEDIA_TYPE.value
+
+
+class RangeNotSatisfiable(LambdaFWException):
+    """416 - Cannot satisfy request range"""
+
+    message = HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE.description
+    status_code = HTTPStatus.REQUESTED_RANGE_NOT_SATISFIABLE.value
+
+
+class ExpectationFailed(LambdaFWException):
+    """417 - Expect condition could not be satisfied"""
+
+    message = HTTPStatus.EXPECTATION_FAILED.description
+    status_code = HTTPStatus.EXPECTATION_FAILED.value
+
+
+class ImATeapot(LambdaFWException):
+    """418 - The server refuses the attempt to brew coffee with a teapot"""
+
+    message = "The server refuses the attempt to brew coffee with a teapot"
+    status_code = 418
+
+
+class MisdirectedRequest(LambdaFWException):
+    """421 - Server is not able to produce a response"""
+
+    message = HTTPStatus.MISDIRECTED_REQUEST.description
+    status_code = HTTPStatus.MISDIRECTED_REQUEST.value
+
+
+class UnprocessableEntity(LambdaFWException):
+    """422 -"""
+
+    message = HTTPStatus.UNPROCESSABLE_ENTITY.description
+    status_code = HTTPStatus.UNPROCESSABLE_ENTITY.value
+
+
+class Locked(LambdaFWException):
+    """423 -"""
+
+    message = HTTPStatus.LOCKED.description
+    status_code = HTTPStatus.LOCKED.value
+
+
+class FailedDependency(LambdaFWException):
+    """424 -"""
+
+    message = HTTPStatus.FAILED_DEPENDENCY.description
+    status_code = HTTPStatus.FAILED_DEPENDENCY.value
+
+
+class UpgradeRequired(LambdaFWException):
+    """426 -"""
+
+    message = HTTPStatus.UPGRADE_REQUIRED.description
+    status_code = HTTPStatus.UPGRADE_REQUIRED.value
+
+
+class PreconditionRequired(LambdaFWException):
+    """428 - The origin server requires the request to be conditional"""
+
+    message = HTTPStatus.PRECONDITION_REQUIRED.description
+    status_code = HTTPStatus.PRECONDITION_REQUIRED.value
+
+
+class TooManyRequests(LambdaFWException):
+    """429 - The user has sent too many requests in a given amount of time ("rate limiting")"""
+
+    message = HTTPStatus.TOO_MANY_REQUESTS.description
+    status_code = HTTPStatus.TOO_MANY_REQUESTS.value
+
+
+class RequestHeaderFieldsTooLarge(LambdaFWException):
+    """431 - The server is unwilling to process the request because its header fields are too large"""
+
+    message = HTTPStatus.REQUEST_HEADER_FIELDS_TOO_LARGE.description
+    status_code = HTTPStatus.REQUEST_HEADER_FIELDS_TOO_LARGE.value
+
+
+class UnavailableForLegalReasons(LambdaFWException):
+    """451 - The server is denying access to the resource as a consequence of a legal demand"""
+
+    message = HTTPStatus.UNAVAILABLE_FOR_LEGAL_REASONS.description
+    status_code = HTTPStatus.UNAVAILABLE_FOR_LEGAL_REASONS.value
+
+
+class ServerError(LambdaFWException):
+    """500 - Server got itself in trouble"""
+
+    message = HTTPStatus.INTERNAL_SERVER_ERROR.description
+    status_code = HTTPStatus.INTERNAL_SERVER_ERROR.value
+
+
+class NotImplementedHTTP(LambdaFWException):
```

Comment:
That is the only exception with an _HTTP_ suffix 🤔  Do we need to add that here?

Reply:
Yest as we would overwrite NotImplemented builtin

Reply:
It makes a lot of sense! So maybe we can name it e.g. [NotImplementedFunctionality](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501)? Just an idea but this HTTP sounds like it should be put everywhere or nowhere 😉